### PR TITLE
Fully completed integration torpedo with GUI.

### PR DESCRIPTION
**Pull request by paulstarbi**
Server torpedo logic integrated with GUI.
Added Locale resolver to torpedoes left label.

### CODE REVIEW

File: src/main/java/com/fourmusketeers/client/logic/ShipsContainer.java [hunk 1]
```diff
@@ -18,6 +18,18 @@
...
+
+        ships.forEach((ship) -> {
+            if (ship.getMaxTilesSize() == 1) {
+                oneMastShips.add(ship);
+            }
+        });
+
+        return oneMastShips;
+    }
+
```
Comment (kukulastudent) [inline src/main/java/com/fourmusketeers/client/logic/ShipsContainer.java]: lambda :+1:

File: src/main/java/com/fourmusketeers/client/view/GameWindowView.java [hunk 2]
```diff
@@ -166,19 +185,67 @@ private void updateBoard(GameTurnInfoToClient infoPackage) {
         });
     }
 
+    private void updateTorpedoesView() {
+        Platform.runLater(() -> torpedoesLeftLabel.setText("" + torpedoesLeft));
```
Comment (kukulastudent) [inline src/main/java/com/fourmusketeers/client/view/GameWindowView.java]: :+1:

File: src/main/resources/localizationPL.properties [hunk 3]
```diff
@@ -10,4 +10,5 @@ fourMast=Czteromasztowiec:
 threeMast=Trzymasztowce:
 twoMast=Dwumasztowce:
 oneMast=Jednomasztowce:
-random=Losowe rozmieszczenie statk\u00F3w
\ No newline at end of file
+random=Losowe rozmieszczenie statk\u00F3w
```
Comment (Dzionek95) [inline src/main/resources/localizationPL.properties]: 👍

File: src/main/java/com/fourmusketeers/client/view/GameWindowView.java [hunk 4]
```diff
@@ -166,19 +185,67 @@ private void updateBoard(GameTurnInfoToClient infoPackage) {
...
+
+    private boolean isTorpedoesLeft() {
+        return torpedoesLeft > 0;
+    }
+
+    private void setTorpedoesLeft(GameTurnInfoToClient torpedoesLeft) {
+        this.torpedoesLeft = torpedoesLeft.getTorpedoesLeftCount();
+    }
+
+    private HBox getTorpedoesLeftView() {
```
Comment (kukulastudent) [inline src/main/java/com/fourmusketeers/client/view/GameWindowView.java]: HBO Game of thrones? :D

File: src/main/java/com/fourmusketeers/client/view/GameWindowView.java [hunk 5]
```diff
@@ -145,17 +156,25 @@ private void setupWhoStarts(WhoStarts whoStarts) {
...
+                enemyButtons.get(finalDetonationPosition).getButton().setStyle("-fx-background-color: red");
+                fillTorpedoTrajectory(infoPackage, enemyButtons);
             } else if (!infoPackage.isOnTarget() && infoPackage.canIPlay()) { //Opposite player missed
-                myButtons.get(shotPosition).getButton().setText("X");
+                myButtons.get(finalDetonationPosition).getButton().setText("X");
+                fillTorpedoTrajectory(infoPackage, myButtons);
             } else if (infoPackage.isOnTarget() && !infoPackage.canIPlay()) { //I got shot
-                myButtons.get(shotPosition).getButton().setStyle("-fx-background-color: red");
+                myButtons.get(finalDetonationPosition).getButton().setStyle("-fx-background-color: red");
+                fillTorpedoTrajectory(infoPackage, myButtons);
```
Comment (Dzionek95) [inline src/main/java/com/fourmusketeers/client/view/GameWindowView.java]: cool

File: src/main/java/com/fourmusketeers/client/logic/ShipsContainer.java (hunk 1 above)
Comment (Dzionek95) [inline src/main/java/com/fourmusketeers/client/logic/ShipsContainer.java]: 💛

**Pull request merged**